// @atomist/rug/operations/Decorators.js
"use strict";
//used by annotation functions
function set_metadata(obj, key, value) {
    Object.defineProperty(obj, key, { value: value, writable: false, enumerable: true });
}
function get_metadata(obj, key) {
    var desc = Object.getOwnPropertyDescriptor(obj, key);
    if ((desc == null || desc == undefined) && (obj.prototype != undefined)) {
        desc = Object.getOwnPropertyDescriptor(obj.prototype, key);
    }
    if (desc != null || desc != undefined) {
        return desc.value;
    }
    return null;
}
/**
* Decorator for parameters. Adds to object properties
*/
function Parameter(details) {
    return function (target, propertyKey) {
        var params = get_metadata(target, "__parameters");
        if (params == null) {
            params = [];
        }
        details["name"] = propertyKey;
        details["decorated"] = true;
        params.push(details);
        set_metadata(target, "__parameters", params);
    };
}
exports.Parameter = Parameter;
/**
* Decorator for editors. Use this intead of implementing the editor interface.
*/
function ruglike(fn, kind, msg) {
    return function (name, description) {
        return function (ctr) {
            if (typeof ctr.prototype[fn] !== "function") {
                throw new Error("" + msg);
            }
            ctr.prototype["__name"] = name;
            ctr.prototype["__description"] = description;
            ctr.prototype["__kind"] = kind;
        };
    };
}
var Generator = ruglike("populate", "generator", "populate must be a function with first parameter = Project");
exports.Generator = Generator;
var Reviewer = ruglike("review", "reviewer", "review must be a function with first parameter = Project");
exports.Reviewer = Reviewer;
var Editor = ruglike("edit", "editor", "edit must be a function with first parameter = Project");
exports.Editor = Editor;
var CommandHandler = ruglike("handle", "command-handler", "handle must be a function with first parameter = CommandContext");
exports.CommandHandler = CommandHandler;
var ResponseHandler = ruglike("handle", "response-handler", "handle must be a function with first paramter = Response<T>");
exports.ResponseHandler = ResponseHandler;
var EventHandler = function (name, description, expression) {
    return function (ctr) {
        if (typeof ctr.prototype["handle"] !== "function") {
            throw new Error("handle must be a function with first parameter = Match<R,N>");
        }
        ctr.prototype["__name"] = name;
        ctr.prototype["__description"] = description;
        ctr.prototype["__kind"] = "event-handler";
        if (typeof expression == 'string') {
            ctr.prototype["__expression"] = expression;
        }
        else {
            ctr.prototype["__expression"] = expression.expression;
        }
    };
};
exports.EventHandler = EventHandler;
/**
* Decorator for tags. Sets tags on the class
*/
function Tags() {
    var tags = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        tags[_i] = arguments[_i];
    }
    return function (target) {
        target.prototype["__tags"] = tags;
    };
}
exports.Tags = Tags;
function Intent() {
    var intent = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        intent[_i] = arguments[_i];
    }
    return function (target) {
        target.prototype["__intent"] = intent;
    };
}
exports.Intent = Intent;
