// @atomist/rug/operations/RugOperation.js
"use strict";
var Pattern = (function () {
    function Pattern() {
    }
    return Pattern;
}());
Pattern.url = "@java_package";
Pattern.any = "@any";
Pattern.group_id = "@group_id";
Pattern.java_class = "@java_class";
Pattern.java_identifier = "@java_identifier";
Pattern.java_package = "@java_package";
Pattern.project_name = "@project_name";
Pattern.port = "@port";
Pattern.ruby_class = "@ruby_class";
Pattern.ruby_identifier = "@ruby_identifier";
Pattern.semantic_version = "@semantic_version";
Pattern.uuid = "@uuid";
exports.Pattern = Pattern;
/**
 * Status of an operation.
 */
var Status;
(function (Status) {
    Status[Status["Success"] = 0] = "Success";
    Status[Status["NoChange"] = 1] = "NoChange";
    Status[Status["Error"] = 2] = "Error";
})(Status || (Status = {}));
exports.Status = Status;
/**
 * Result of running an editor
 */
var Result = (function () {
    function Result(status, message) {
        if (message === void 0) { message = ""; }
        this.status = status;
        this.message = message;
    }
    return Result;
}());
exports.Result = Result;
