// @atomist/rug/operations/RugOperation.js
"use strict";
/**
 * Well-known patterns used in operation parameters.
 */
var Pattern = (function () {
    function Pattern() {
    }
    return Pattern;
}());
Pattern.url = "@java_package";
Pattern.any = "@any";
Pattern.group_id = "@group_id";
Pattern.java_class = "@java_class";
Pattern.java_identifier = "@java_identifier";
Pattern.java_package = "@java_package";
Pattern.project_name = "@project_name";
Pattern.port = "@port";
Pattern.ruby_class = "@ruby_class";
Pattern.ruby_identifier = "@ruby_identifier";
Pattern.semantic_version = "@semantic_version";
Pattern.version_range = "@version_range";
Pattern.uuid = "@uuid";
exports.Pattern = Pattern;
/**
 * Status of an operation.
 */
var Status;
(function (Status) {
    Status[Status["Success"] = 0] = "Success";
    Status[Status["NoChange"] = 1] = "NoChange";
    Status[Status["Error"] = 2] = "Error";
})(Status || (Status = {}));
exports.Status = Status;
/**
 * Result of running an editor
 */
var Result = (function () {
    function Result(status, message) {
        if (message === void 0) { message = ""; }
        this.status = status;
        this.message = message;
    }
    return Result;
}());
exports.Result = Result;
/**
 * Severity of a comment from a review
 */
var Severity;
(function (Severity) {
    Severity[Severity["Fine"] = 0] = "Fine";
    Severity[Severity["Polish"] = 1] = "Polish";
    Severity[Severity["Major"] = 2] = "Major";
    Severity[Severity["Broken"] = 3] = "Broken";
})(Severity || (Severity = {}));
exports.Severity = Severity;
/**
 * A comment from a review
 */
var ReviewComment = (function () {
    function ReviewComment(comment, severity, fileName, line, column) {
        this.comment = comment;
        this.severity = severity;
        this.fileName = fileName;
        this.line = line;
        this.column = column;
    }
    return ReviewComment;
}());
exports.ReviewComment = ReviewComment;
/**
 * Result of running a reviewer
 */
var ReviewResult = (function () {
    function ReviewResult(note, comments) {
        this.note = note;
        this.comments = comments;
    }
    return ReviewResult;
}());
exports.ReviewResult = ReviewResult;
