// @atomist/cortex/stub/ChatChannel.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type ChatChannel
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var ChatChannel = (function () {
    function ChatChannel() {
    }
    ChatChannel.prototype.nodeName = function () {
        return "ChatChannel";
    };
    ChatChannel.prototype.nodeTags = function () {
        return ["ChatChannel", "-dynamic"];
    };
    /**
      * discusses - ChatChannel -> Repo
      *
      * @returns {Repo[]}
      */
    ChatChannel.prototype.discusses = function () {
        return this._discusses;
    };
    ChatChannel.prototype.withDiscusses = function (discusses) {
        this._discusses = discusses;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    ChatChannel.prototype.id = function () {
        return this._id;
    };
    ChatChannel.prototype.withId = function (id) {
        this._id = id;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    ChatChannel.prototype.name = function () {
        return this._name;
    };
    ChatChannel.prototype.withName = function (name) {
        this._name = name;
        return this;
    };
    return ChatChannel;
}());
exports.ChatChannel = ChatChannel;
