// @atomist/cortex/stub/Delta.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Delta
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Delta = (function () {
    function Delta() {
    }
    Delta.prototype.nodeName = function () {
        return "Delta";
    };
    Delta.prototype.nodeTags = function () {
        return ["Delta", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    Delta.prototype.path = function () {
        return this._path;
    };
    Delta.prototype.withPath = function (path) {
        this._path = path;
        return this;
    };
    return Delta;
}());
exports.Delta = Delta;
