// @atomist/cortex/stub/GitHubId.ts
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { GraphNode } from "@atomist/rug/tree/PathExpression";
import * as api from "../GitHubId";
import { Person } from "./Person";
export { GitHubId };

/*
 * Type GitHubId
 * 
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
class GitHubId implements api.GitHubId {

    private _login: string;
    private _name: string;
    private _of: Person;

    nodeName(): string {
        return "GitHubId";
    }

    nodeTags(): string[] {
        return [ "GitHubId", "-dynamic" ];
    }

    /**
      * String
      *
      * @returns {string}
      */
    login(): string {
        return this._login;
    }

    withLogin(login: string): GitHubId {
        this._login = login;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    name(): string {
        return this._name;
    }

    withName(name: string): GitHubId {
        this._name = name;
        return this;
    }

    /**
      * of - GitHubId -> Person
      *
      * @returns {Person}
      */
    of(): Person {
        return this._of;
    }

    withOf(of: Person): GitHubId {
        this._of = of;
        return this;
    }

}
