// @atomist/cortex/stub/Org.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Org
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Org = (function () {
    function Org() {
    }
    Org.prototype.nodeName = function () {
        return "Org";
    };
    Org.prototype.nodeTags = function () {
        return ["Org", "-dynamic"];
    };
    /**
      * has - Org -> Repo
      *
      * @returns {Repo[]}
      */
    Org.prototype.has = function () {
        return this._has;
    };
    Org.prototype.withHas = function (has) {
        this._has = has;
        return this;
    };
    /**
      * ownedBy - Org -> ChatTeam
      *
      * @returns {ChatTeam}
      */
    Org.prototype.ownedBy = function () {
        return this._ownedBy;
    };
    Org.prototype.withOwnedBy = function (ownedBy) {
        this._ownedBy = ownedBy;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Org.prototype.owner = function () {
        return this._owner;
    };
    Org.prototype.withOwner = function (owner) {
        this._owner = owner;
        return this;
    };
    return Org;
}());
exports.Org = Org;
