// @atomist/cortex/stub/Release.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Release
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Release = (function () {
    function Release() {
    }
    Release.prototype.nodeName = function () {
        return "Release";
    };
    Release.prototype.nodeTags = function () {
        return ["Release", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    Release.prototype.id = function () {
        return this._id;
    };
    Release.prototype.withId = function (id) {
        this._id = id;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Release.prototype.name = function () {
        return this._name;
    };
    Release.prototype.withName = function (name) {
        this._name = name;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Release.prototype.timestamp = function () {
        return this._timestamp;
    };
    Release.prototype.withTimestamp = function (timestamp) {
        this._timestamp = timestamp;
        return this;
    };
    return Release;
}());
exports.Release = Release;
