// @atomist/cortex/stub/Spec.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Spec
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Spec = (function () {
    function Spec() {
    }
    Spec.prototype.nodeName = function () {
        return "Spec";
    };
    Spec.prototype.nodeTags = function () {
        return ["Spec", "-dynamic"];
    };
    /**
      * creates - Spec -> Pod
      *
      * @returns {Pod[]}
      */
    Spec.prototype.creates = function () {
        return this._creates;
    };
    Spec.prototype.withCreates = function (creates) {
        this._creates = creates;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Spec.prototype.curHash = function () {
        return this._curHash;
    };
    Spec.prototype.withCurHash = function (curHash) {
        this._curHash = curHash;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Spec.prototype.fsha = function () {
        return this._fsha;
    };
    Spec.prototype.withFsha = function (fsha) {
        this._fsha = fsha;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Spec.prototype.kind = function () {
        return this._kind;
    };
    Spec.prototype.withKind = function (kind) {
        this._kind = kind;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Spec.prototype.name = function () {
        return this._name;
    };
    Spec.prototype.withName = function (name) {
        this._name = name;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Spec.prototype.org = function () {
        return this._org;
    };
    Spec.prototype.withOrg = function (org) {
        this._org = org;
        return this;
    };
    /**
      * references - Spec -> Container
      *
      * @returns {Container[]}
      */
    Spec.prototype.references = function () {
        return this._references;
    };
    Spec.prototype.withReferences = function (references) {
        this._references = references;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Spec.prototype.repo = function () {
        return this._repo;
    };
    Spec.prototype.withRepo = function (repo) {
        this._repo = repo;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Spec.prototype.timestamp = function () {
        return this._timestamp;
    };
    Spec.prototype.withTimestamp = function (timestamp) {
        this._timestamp = timestamp;
        return this;
    };
    return Spec;
}());
exports.Spec = Spec;
