// @atomist/cortex/stub/Stage.ts
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { GraphNode } from "@atomist/rug/tree/PathExpression";
import * as api from "../Stage";

export { Stage };

/*
 * Type Stage
 * 
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
class Stage implements api.Stage {

    private _endTime: string;
    private _id: string;
    private _name: string;
    private _refId: string;
    private _startTime: string;
    private _status: string;
    private _type: string;

    nodeName(): string {
        return "Stage";
    }

    nodeTags(): string[] {
        return [ "Stage", "-dynamic" ];
    }

    /**
      * String
      *
      * @returns {string}
      */
    endTime(): string {
        return this._endTime;
    }

    withEndTime(endTime: string): Stage {
        this._endTime = endTime;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    id(): string {
        return this._id;
    }

    withId(id: string): Stage {
        this._id = id;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    name(): string {
        return this._name;
    }

    withName(name: string): Stage {
        this._name = name;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    refId(): string {
        return this._refId;
    }

    withRefId(refId: string): Stage {
        this._refId = refId;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    startTime(): string {
        return this._startTime;
    }

    withStartTime(startTime: string): Stage {
        this._startTime = startTime;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    status(): string {
        return this._status;
    }

    withStatus(status: string): Stage {
        this._status = status;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    type(): string {
        return this._type;
    }

    withType(type: string): Stage {
        this._type = type;
        return this;
    }

}
