// @atomist/cortex/stub/Container.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Container
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Container = (function () {
    function Container() {
    }
    Container.prototype.nodeName = function () {
        return "Container";
    };
    Container.prototype.nodeTags = function () {
        return ["Container", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    Container.prototype.image = function () {
        return this._image;
    };
    Container.prototype.withImage = function (image) {
        this._image = image;
        return this;
    };
    /**
      * isTagged - Container -> Tag
      *
      * @returns {Tag}
      */
    Container.prototype.isTagged = function () {
        return this._isTagged;
    };
    Container.prototype.withIsTagged = function (isTagged) {
        this._isTagged = isTagged;
        return this;
    };
    /**
      * referencedBy - Container -> Spec
      *
      * @returns {Spec}
      */
    Container.prototype.referencedBy = function () {
        return this._referencedBy;
    };
    Container.prototype.withReferencedBy = function (referencedBy) {
        this._referencedBy = referencedBy;
        return this;
    };
    return Container;
}());
exports.Container = Container;
