// @atomist/cortex/stub/Email.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Email
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Email = (function () {
    function Email() {
    }
    Email.prototype.nodeName = function () {
        return "Email";
    };
    Email.prototype.nodeTags = function () {
        return ["Email", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    Email.prototype.address = function () {
        return this._address;
    };
    Email.prototype.withAddress = function (address) {
        this._address = address;
        return this;
    };
    /**
      * of - Email -> Person
      *
      * @returns {Person}
      */
    Email.prototype.of = function () {
        return this._of;
    };
    Email.prototype.withOf = function (of) {
        this._of = of;
        return this;
    };
    return Email;
}());
exports.Email = Email;
