// @atomist/cortex/stub/Environment.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Environment
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Environment = (function () {
    function Environment() {
    }
    Environment.prototype.nodeName = function () {
        return "Environment";
    };
    Environment.prototype.nodeTags = function () {
        return ["Environment", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    Environment.prototype.domainName = function () {
        return this._domainName;
    };
    Environment.prototype.withDomainName = function (domainName) {
        this._domainName = domainName;
        return this;
    };
    /**
      * owns - Environment -> Spec
      *
      * @returns {Spec}
      */
    Environment.prototype.owns = function () {
        return this._owns;
    };
    Environment.prototype.withOwns = function (owns) {
        this._owns = owns;
        return this;
    };
    return Environment;
}());
exports.Environment = Environment;
