// @atomist/cortex/stub/GitHubId.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type GitHubId
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var GitHubId = (function () {
    function GitHubId() {
    }
    GitHubId.prototype.nodeName = function () {
        return "GitHubId";
    };
    GitHubId.prototype.nodeTags = function () {
        return ["GitHubId", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    GitHubId.prototype.login = function () {
        return this._login;
    };
    GitHubId.prototype.withLogin = function (login) {
        this._login = login;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    GitHubId.prototype.name = function () {
        return this._name;
    };
    GitHubId.prototype.withName = function (name) {
        this._name = name;
        return this;
    };
    /**
      * of - GitHubId -> Person
      *
      * @returns {Person}
      */
    GitHubId.prototype.of = function () {
        return this._of;
    };
    GitHubId.prototype.withOf = function (of) {
        this._of = of;
        return this;
    };
    return GitHubId;
}());
exports.GitHubId = GitHubId;
