// @atomist/cortex/stub/Heroku.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Heroku
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Heroku = (function () {
    function Heroku() {
    }
    Heroku.prototype.nodeName = function () {
        return "Heroku";
    };
    Heroku.prototype.nodeTags = function () {
        return ["Heroku", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    Heroku.prototype.app = function () {
        return this._app;
    };
    Heroku.prototype.withApp = function (app) {
        this._app = app;
        return this;
    };
    /**
      * deploys - Heroku -> Commit
      *
      * @returns {Commit[]}
      */
    Heroku.prototype.deploys = function () {
        return this._deploys;
    };
    Heroku.prototype.withDeploys = function (deploys) {
        this._deploys = deploys;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Heroku.prototype.release = function () {
        return this._release;
    };
    Heroku.prototype.withRelease = function (release) {
        this._release = release;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Heroku.prototype.timestamp = function () {
        return this._timestamp;
    };
    Heroku.prototype.withTimestamp = function (timestamp) {
        this._timestamp = timestamp;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Heroku.prototype.url = function () {
        return this._url;
    };
    Heroku.prototype.withUrl = function (url) {
        this._url = url;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Heroku.prototype.user = function () {
        return this._user;
    };
    Heroku.prototype.withUser = function (user) {
        this._user = user;
        return this;
    };
    return Heroku;
}());
exports.Heroku = Heroku;
