// @atomist/cortex/stub/Label.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Label
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Label = (function () {
    function Label() {
    }
    Label.prototype.nodeName = function () {
        return "Label";
    };
    Label.prototype.nodeTags = function () {
        return ["Label", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    Label.prototype.id = function () {
        return this._id;
    };
    Label.prototype.withId = function (id) {
        this._id = id;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Label.prototype.name = function () {
        return this._name;
    };
    Label.prototype.withName = function (name) {
        this._name = name;
        return this;
    };
    return Label;
}());
exports.Label = Label;
