// @atomist/cortex/stub/Repo.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Repo
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Repo = (function () {
    function Repo() {
    }
    Repo.prototype.nodeName = function () {
        return "Repo";
    };
    Repo.prototype.nodeTags = function () {
        return ["Repo", "-dynamic"];
    };
    /**
      * channel - Repo -> ChatChannel
      *
      * @returns {ChatChannel[]}
      */
    Repo.prototype.channel = function () {
        return this._channel;
    };
    Repo.prototype.withChannel = function (channel) {
        this._channel = channel;
        return this;
    };
    /**
      * hasBuilds - Repo -> Build
      *
      * @returns {Build[]}
      */
    Repo.prototype.hasBuilds = function () {
        return this._hasBuilds;
    };
    Repo.prototype.withHasBuilds = function (hasBuilds) {
        this._hasBuilds = hasBuilds;
        return this;
    };
    /**
      * hasIssues - Repo -> Issue
      *
      * @returns {Issue[]}
      */
    Repo.prototype.hasIssues = function () {
        return this._hasIssues;
    };
    Repo.prototype.withHasIssues = function (hasIssues) {
        this._hasIssues = hasIssues;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Repo.prototype.name = function () {
        return this._name;
    };
    Repo.prototype.withName = function (name) {
        this._name = name;
        return this;
    };
    /**
      * ownedBy - Repo -> Org
      *
      * @returns {Org}
      */
    Repo.prototype.ownedBy = function () {
        return this._ownedBy;
    };
    Repo.prototype.withOwnedBy = function (ownedBy) {
        this._ownedBy = ownedBy;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Repo.prototype.owner = function () {
        return this._owner;
    };
    Repo.prototype.withOwner = function (owner) {
        this._owner = owner;
        return this;
    };
    /**
      * uses - Repo -> Label
      *
      * @returns {Label[]}
      */
    Repo.prototype.uses = function () {
        return this._uses;
    };
    Repo.prototype.withUses = function (uses) {
        this._uses = uses;
        return this;
    };
    return Repo;
}());
exports.Repo = Repo;
