// @atomist/cortex/stub/Spec.ts
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { GraphNode } from "@atomist/rug/tree/PathExpression";
import * as api from "../Spec";
import { Container } from "./Container";
import { Pod } from "./Pod";
export { Spec };

/*
 * Type Spec
 * 
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
class Spec implements api.Spec {

    private _creates: Pod[];
    private _curHash: string;
    private _fsha: string;
    private _kind: string;
    private _name: string;
    private _org: string;
    private _references: Container[];
    private _repo: string;
    private _timestamp: string;

    nodeName(): string {
        return "Spec";
    }

    nodeTags(): string[] {
        return [ "Spec", "-dynamic" ];
    }

    /**
      * creates - Spec -> Pod
      *
      * @returns {Pod[]}
      */
    creates(): Pod[] {
        return this._creates;
    }

    withCreates(creates: Pod[]): Spec {
        this._creates = creates;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    curHash(): string {
        return this._curHash;
    }

    withCurHash(curHash: string): Spec {
        this._curHash = curHash;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    fsha(): string {
        return this._fsha;
    }

    withFsha(fsha: string): Spec {
        this._fsha = fsha;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    kind(): string {
        return this._kind;
    }

    withKind(kind: string): Spec {
        this._kind = kind;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    name(): string {
        return this._name;
    }

    withName(name: string): Spec {
        this._name = name;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    org(): string {
        return this._org;
    }

    withOrg(org: string): Spec {
        this._org = org;
        return this;
    }

    /**
      * references - Spec -> Container
      *
      * @returns {Container[]}
      */
    references(): Container[] {
        return this._references;
    }

    withReferences(references: Container[]): Spec {
        this._references = references;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    repo(): string {
        return this._repo;
    }

    withRepo(repo: string): Spec {
        this._repo = repo;
        return this;
    }

    /**
      * String
      *
      * @returns {string}
      */
    timestamp(): string {
        return this._timestamp;
    }

    withTimestamp(timestamp: string): Spec {
        this._timestamp = timestamp;
        return this;
    }

}
