// @atomist/cortex/stub/Status.js
/*
 * Copyright 2015-2017 Atomist Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
"use strict";
/*
 * Type Status
 *
 * Generated class exposing Atomist Cortex.
 * Fluent builder style class for use in testing and query by example.
 */
var Status = (function () {
    function Status() {
    }
    Status.prototype.nodeName = function () {
        return "Status";
    };
    Status.prototype.nodeTags = function () {
        return ["Status", "-dynamic"];
    };
    /**
      * String
      *
      * @returns {string}
      */
    Status.prototype.description = function () {
        return this._description;
    };
    Status.prototype.withDescription = function (description) {
        this._description = description;
        return this;
    };
    /**
      * on - Status -> Commit
      *
      * @returns {Commit}
      */
    Status.prototype.on = function () {
        return this._on;
    };
    Status.prototype.withOn = function (on) {
        this._on = on;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Status.prototype.state = function () {
        return this._state;
    };
    Status.prototype.withState = function (state) {
        this._state = state;
        return this;
    };
    /**
      * String
      *
      * @returns {string}
      */
    Status.prototype.targetUrl = function () {
        return this._targetUrl;
    };
    Status.prototype.withTargetUrl = function (targetUrl) {
        this._targetUrl = targetUrl;
        return this;
    };
    return Status;
}());
exports.Status = Status;
